tests/lib/tools/mkimage-uc22

```diff
@@ -0,0 +1,564 @@
+# run outside of spread easily for quick interactive testing
+
+DEFAULT_BASE_DIR="/tmp/mkimage"
+DEFAULT_BOOT_IMG="boot.img"
+DEFAULT_CACHE_DIR="cache"
+DEFAULT_MNT_DIR="mnt"
+DEFAULT_SNAPD_BINPATH="/usr/lib/snapd"
+DEFAULT_BASE_IMAGE_URL="http://cdimage.ubuntu.com/ubuntu-base/releases/22.04/release/ubuntu-base-22.04-base-amd64.tar.gz"
+DEFAULT_SNAPD_DEB_URL=https://storage.googleapis.com/snapd-spread-tests/snapd-tests/packages/ubuntu-22.04-64.tar 
+
+show_help() {
+    echo "usage: mkimage-uc22 [--boot-img BOOT_IMG] [--cache-dir CACHE_DIR] [--mnt-dir MNT_DIR] [--snap <SNAP_PATH> ...] model_assertion" 
+    echo ""
+    echo "create an image that looks like a classic image"
+    echo ""
+    echo "Available options:"
+    echo "  -h --help   show this help message."
+    echo ""
+}
+
+find_snap(){
+    local SNAPS_DIR="$1"
+    local SNAP="$2"
+    find "$SNAPS_DIR" -maxdepth 1 -name "${SNAP}_*.snap" -printf "%f\n"
+}
+
+get_extra_snaps() {
+    local SNAPS_DIR="$1"
+    local EXTRA_SNAPS=""
+
+    if [ -d "$CACHE_DIR" ]; then
+        while IFS= read -r mysnap; do
+            echo "$mysnap"
+        done < <(find "$SNAPS_DIR" -name '*.snap')
+    fi
+}
+
+get_assets() {
+    CACHE="$1"
+    BASE_IMAGE_URL="$2"
+    BASE_IMAGE_PATH="$3"
+    shift 3
+    # shellcheck disable=SC2124
+    SNAPS="$@"
+
+    if [ -d "$CACHE" ]; then
+        echo "mkimage-uc22: using existing cache dir $CACHE"
+    fi
+
+    for snap in $SNAPS; do
+        if [ ! -f "$snap" ]; then
+            echo "mkimage-uc22: the provided snap \"$snap\" does not exist" >&2
+            return 1
+        fi
+    done
+
+    mkdir -p "$CACHE"
+    # shellcheck disable=SC2086
+    cp $SNAPS "$CACHE"
+
+    # get the snaps
+    for snap in pc-kernel pc; do        
+        NAME="$(find_snap "$CACHE" "$snap")"
+        if [ -z "$NAME" ]; then
+            echo "mkimage-uc22: downloading snap $snap"
+            snap download --channel=22 --target-directory="$CACHE" "$snap"
+        else
+            echo "mkimage-uc22: using provided snap $NAME"
+        fi
+        unsquashfs -n -d "$CACHE"/snap-"$snap" "$CACHE"/"${snap}"_*.snap
+    done
+    for snap in snapd core22; do
+        NAME="$(find_snap "$CACHE" "$snap")"
+        if [ -z "$NAME" ]; then
+            echo "mkimage-uc22: downloading snap $snap"
+            snap download --target-directory="$CACHE" "$snap"
+        else
+            echo "mkimage-uc22: using provided snap $NAME"
+        fi
+    done
+
+    if [ -n "$BASE_IMAGE_PATH" ]; then
+        echo "mkimage-uc22: using local ubuntu classic base from $BASE_IMAGE_PATH"
+        cp "$BASE_IMAGE_PATH" "$CACHE"
+    else
+        echo "mkimage-uc22: downloading the ubuntu classic base"
+        wget -c -P "$CACHE" "$BASE_IMAGE_URL"
+    fi    
+
+    echo "mkimage-uc22: image assets ready"
+}
+
+# shellcheck disable=SC2317
+cleanup() {
+    IMG="$(readlink -f "$1")"
+    MNT="$(readlink -f "$2")"
+
+    sleep 1
+    sudo umount "$MNT"/* || true
```

Comment:
would it make sense to retry 2 or 3 times instead of sleeping?

Reply:
I did the change, also removed the '|| true' which is very dangerous